### PR TITLE
feat(ui): bump frontend dependencies

### PR DESCRIPTION
**Pull request by ngamanda**
There's a whole bunch of lint warnings now. I'm setting them to `warn` for your review. Let me know if you'd prefer to turn them off or just bulk update them.

* Bump dependencies 
  * Cleaned up duplicated packages in `yarn.lock`
  * Migrated `@vs` packages to `@visx`
  * Remove implementation for `@data-ui/xy-chart` to `@visx/xychart`
* Fix ESLint config, VSCode now shows all lint errors
  * Removed unused plugins
  * Updated AirBnB plugins
  * Disabled/warn the newly added/changed rules
* Bump Gradle node version to LTS
* Remove 'react-email-share-link'
  * Had some trouble compiling it, so I opted to directly copy the implementation over instead
* Configure jest to use `moduleNameMapper` to map packages rather than transpiling them again

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

**Comment by joshuaeilers** [comment]
Hey @ngamanda - so sorry for the delay on this review. I'm pulling down the code now and going to do a little test drive with it and then start reviewing shortly after. There's a coupe conflicts, but I'll do my local testing rebased on master.

File: datahub-web-react/.eslintrc.js
Comment (joshuaeilers) [inline datahub-web-react/.eslintrc.js:21]: I think my preference here would be to try to make this upgrade as transparent as possible, for now. While it's more ideal to fix a lot of these, doing this in stages where we upgrade with the warnings=off, and then can address one at a time. Helps keep the changeset a bit smaller.

File: datahub-web-react/build.gradle
Comment (joshuaeilers) [inline datahub-web-react/build.gradle:19]: Thank you so much for this!

File: datahub-web-react/craco.config.js
Comment (joshuaeilers) [inline datahub-web-react/craco.config.js:1]: Nice, this kinda stuff for webpack upgrades is really tricky to figure out 🙏

File: datahub-web-react/src/app/shared/share/items/EmailMenuItem.tsx
Comment (joshuaeilers) [inline datahub-web-react/src/app/shared/share/items/EmailMenuItem.tsx:5]: Yeah this component doesn't even work for me on our datahub demo site... seems like you have parity with the react-email-share-link functionality though.

Comment (joshuaeilers) [reply datahub-web-react/src/app/shared/share/items/EmailMenuItem.tsx:5]: Ah, it's probably because I didn't set my mail client. When I did mac mail, it worked. So I think this is all good.

Comment (ngamanda) [reply datahub-web-react/src/app/shared/share/items/EmailMenuItem.tsx:5]: Yea it should be the exact same behaviour since the code was copied as-is. Glad to hear it works!

File: datahub-web-react/.eslintrc.js
Comment (ngamanda) [reply datahub-web-react/.eslintrc.js:21]: Sure, I'll swap all the warnings to off then. You mentioned you tried rebasing locally - will you be updating this MR with the resolved conflicts or should I tackle that instead?

Comment (joshuaeilers) [reply datahub-web-react/.eslintrc.js:21]: I guess more specifically if you could set the warnings to match what the eslintrc had before, that would be great 👍 Not sure if they all really need to be necessarily off if we had some to warn before, thinking to keep parity unless there's a reason we can't do that.

Comment (joshuaeilers) [reply datahub-web-react/.eslintrc.js:21]: I won't be able to directly edit this PR due to permissions, so if you can do that here that would be ideal.

Comment (joshuaeilers) [reply datahub-web-react/.eslintrc.js:21]: The good news is CI passed on my test branch, so I'm gonna plan to do a bit more testing on it today. Then once that's feeling good, we can figure out a time to merge. One thing about this kind of PR is that it can lead to some interruption on developers side so we may want to do a little announcement or something that folks need to re-yarn / upgrade node ideally post-merging this.

Comment (joshuaeilers) [reply datahub-web-react/.eslintrc.js:21]: But we can handle that post-merge of course

**Comment by joshuaeilers** [comment]
@ngamanda I'm noticing the analytics tab's charts x axes values look wrong they appear to be numeric, and I'm guessing it's trying to format a timestamp as a number instead of a date. Could you take a look at that?

**Comment by joshuaeilers** [comment]
Here's a screenshot of what I'm seeing, analytics at the top-right.
Screenshot 2023-09-11 at 9 38 29 AM

**Review by joshuaeilers — CHANGES_REQUESTED** [review]
Setting to changes requested due to the x-axis issue on the charts.

**Comment by ngamanda** [comment]
Hey @joshuaeilers, thanks for catching that! Just committed a fix - lmk if there's any other issues!

**Comment by joshuaeilers** [comment]
This is looking good to me. Tagging in @chriscollins3456 in for the final review and discuss merge timelines. I know we have a couple fairly critical frontend changes coming in this month so we may want to schedule this in October for stability sake, but I'll defer to Chris on that. Thank you again so much for this work!

**Review by joshuaeilers — APPROVED** [review]

File: datahub-web-react/craco.config.js
Comment (chriscollins3456) [inline datahub-web-react/craco.config.js:37]: nice I'm excited about this! hopefully reduces our build size a bit

File: datahub-web-react/.eslintrc.js
Comment (chriscollins3456) [inline datahub-web-react/.eslintrc.js:41]: so these lint changes were necessary as we bumped dependencies we had new linting rules being applied that we didn't want applied?

nothing I'm seeing is concerning to me just wanted to call it out

**Review by chriscollins3456 — APPROVED** [review]
this is looking great! I pulled it down and built everything from scratch and it's looking good to me. the changes make sense and I think this is going to resolve a good amount of our vulnerability issues with older versions of packages as well. thanks so much for this @ngamanda !

I'm going to double check with the team before merging but I think we're close to good to go here

File: datahub-web-react/.eslintrc.js
Comment (ngamanda) [reply datahub-web-react/.eslintrc.js:41]: Yeap! That's right 😃

**Comment by chriscollins3456** [comment]
@ngamanda thank you so much again! just merged in latest and then pulled and ran locally again and we're looking good to go. going to merge now!

**Pull request merged**
